src/events/interactionCreate.ts

```diff
@@ -0,0 +1,39 @@
+import { ButtonInteraction, CommandInteraction, Interaction } from "discord.js";
+import { ctx } from "../ctx";
+import { logger } from "../logger";
+
+export default async function interactionCreate(interaction: Interaction) {
+	try {
+		if (interaction.isCommand()) {
+			await handleCommandInteractions(interaction);
+		} else if (interaction.isButton()) {
+			await handleButtonInteractions(interaction);
+		}
+	} catch (err: unknown) {
+		if (err instanceof Error) {
+			logger.error(`unhandled error occurred on interaction: ${err.name}`);
+			logger.error(err.message);
+			logger.error(err.stack);
+		} else {
+			logger.error(`unknown error occurred ${err}`);
+		}
+	}
+}
+
+async function handleCommandInteractions(interaction: CommandInteraction) {
+	const command = ctx.commands.get(interaction.commandName);
+	if (command) {
+		await command.execute(interaction);
+	} else {
+		logger.error(`error resolving command ${interaction.commandName}`);
+		await interaction.reply("internal error resolving command");
```

Comment:
Big words may confuse user. Also proper puncuation

Reply:
wdym with big words?

Reply:
Try `There was an error running that command. Please contact the developers.`